CommonAssemblyInfo.cs.in

```diff
@@ -24,12 +24,12 @@
 using System.Reflection;
 
 [assembly: AssemblyConfiguration("")]
-[assembly: AssemblyCompany("Jeroen Frijters")]
+[assembly: AssemblyCompany("Windward Studios (based on work by Jeroen Frijters)")]
 [assembly: AssemblyProduct("IKVM.NET")]
-[assembly: AssemblyCopyright("Copyright (C) 2002-2015 Jeroen Frijters")]
+[assembly: AssemblyCopyright("Copyright (C) 2002-2015, 2018 Jeroen Frijters")]
 [assembly: AssemblyTrademark("")]
 [assembly: AssemblyCulture("")]
-[assembly: AssemblyVersion("8.1.@BUILD@.0")]
```

Comment:
I think we still want to keep this, right?

Reply:
What's your reasons? Didn't you change to [manually manage the version](https://github.com/ikvm-revived/ikvm/pull/12/commits/87c4b2b3b9506c018a53bf79526aab906d79d600#diff-0c4d5dd94cbcc94d228b21a02d2dd2dcR32) already?

I think I prefer your approach, because when building the code at some later point without any actual changes, `@BUILD@` leads to unwanted changes. Think of reproducible builds and such stuff, people seem to try removing timestamps etc. Therefore I removed this as well, AFAIK it wasn't used anymore.

Reply:
We do need to make sure the version number gets bumped on any change. @igorkwwrd - any suggestions?

Reply:
I have been using [NerdBank.GitVersioning](https://github.com/AArnott/Nerdbank.GitVersioning) with some success to bump version numbers. It is setup to ensure that a specific git commit always builds the same version number, and you can use branches to determine whether to build a release build or a CI build.

While I am not sure how time consuming it would be to make the change, if you switch the project to the [new `.csproj` format](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj), the `.nuspec` file and `CommonAssemblyInfo.cs` file can be eliminated and most file references can be removed from the project files, which helps to reduce merge conflicts. This format makes accomplishing things like strong naming and optional parameters simpler. It is also the default format that is used when you create a new .NET Standard 2.0 project VS2017 or VS2019.

You are welcome to use whatever you can from the common project structure setup from [J2N](https://github.com/NightOwl888/J2N), [ICU4N](https://github.com/NightOwl888/ICU4N), and [Morfologik.Stemming](https://github.com/NightOwl888/Morfologik.Stemming) as a template to set up this project. The common information for each directory is put into [Build.Directory.props](https://github.com/NightOwl888/ICU4N/blob/master/Directory.Build.props) files, which also link to other common information such as [nuget.props](https://github.com/NightOwl888/ICU4N/blob/master/build/nuget.props), which is where the common NuGet fields are. The [project specific NuGet section](https://github.com/NightOwl888/ICU4N/blob/master/src/ICU4N/ICU4N.csproj#L8-L12) then just requires a few fields.

Reply:
I think this was removed accidentally and it definitely should be kept.  The build number is set on each build as a days difference between now and some day in the past (something like 2000-01-01).  If needed we could change it to some different scheme, e.g. increment by 1.

CommonAssemblyInfo.cs is generating from this .in file.  Currently it is under version control and should be removed.

Reply:
So where to put such a dynamic build number `@BUILD@` then? It doesn't fit into your current scheme of `8.5.0.3`.

Reply:
A good question.
Currently it is updated when NAnt copies .cs.in file into .cs
One way could be to have a file with a version number, let's say version.txt, containing just a version, e.g. 8.5.0.3
In CommonAssemblyInfo.cs.in the version attribute is replaced with the @VERSION@ string.
When copying .cs.in into .cs, the @VERSION@ string gets replaced with the actual value from version.txt

Reply:
And what's the benefit of changing `CommonAssemblyInfo.cs.in` directly? :-) Do you want to generate `version.txt` externally somehow? Additionally, should that `version.txt` contain some dynamic parts, like the past days for `@BUILD@`? Reads to me like somewhat two different problems.

Reply:
I agree, it isn't something that should be resolved right now.

I didn't mean version.txt to be anything special.  Editing CommonAssembly.cs.in directly is even better than having an additional file just for the same purpose.